src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -83,10 +83,9 @@ string DefaultMessageIdStrategy(BasicDeliverEventArgs message)
             return properties.MessageId;
         }
 
-        static Dictionary<string, string> DeserializeHeaders(BasicDeliverEventArgs message)
+        static Dictionary<string, string> DeserializeHeaders(IDictionary<string, object> headers)
```

Comment:
Also, this specific commit is a an additional fixup to be squashed into 0aacc7240952641d5996ac7b4bc90384392b2e7c I realized this morning that I could improve it a bit further by changing the parameter type, hence the fixup commit.